pkg/controller/master/virtualmachine/vm_volume_status_controller.go

```diff
@@ -0,0 +1,493 @@
+package virtualmachine
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	longhornv1 "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	corectl "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtapis "kubevirt.io/client-go/api/v1"
+
+	ctlkubevirtv1 "github.com/harvester/harvester/pkg/generated/controllers/kubevirt.io/v1"
+
+	"github.com/harvester/harvester/pkg/ref"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+const (
+	// only specific LH volume status is checked and updated into virtual machine
+	// all other states are categorized into  "OK"
+	volumeStatusOK                  = volumeStatus("ok")
+	volumeStatusInsufficientStorage = volumeStatus("insufficient storage")
+)
+
+type VMVolumeStatusController struct {
+	vmClient  ctlkubevirtv1.VirtualMachineClient
+	vmCache   ctlkubevirtv1.VirtualMachineCache
+	pvcClient corectl.PersistentVolumeClaimClient
+	pvcCache  corectl.PersistentVolumeClaimCache
+}
+
+type volumeStatus string
+
+type volumeStatusAnnotation struct {
+	// name of volume
+	Name string `json:"name,omitempty"`
+	// abstracted status based on long-horn (LH) volume status
```

Comment:
NIT: long-horn -> longhorn

```suggestion
	// abstracted status based on longhorn (LH) volume status
```

Reply:
Good, will change.